compiler/semantic/op.go

```diff
@@ -192,25 +186,32 @@ func unmarshalHeaders(arena *zed.Arena, val zed.Value) (map[string][]string, err
 	return headers, nil
 }
 
-// XXX At some point SortKey should be an ast.Node but for now just have this
-// return and error and let the parent log the semantic error.
-func semSortKey(p *ast.SortKey) (order.SortKey, error) {
-	if p == nil || p.Keys == nil {
-		return order.Nil, nil
-	}
-	var keys field.List
-	for _, key := range p.Keys {
-		this := DotExprToFieldPath(key)
-		if this == nil {
-			return order.Nil, fmt.Errorf("bad key expr of type %T in file operator", key)
+func (a *analyzer) semSortKeys(sortExprs []ast.SortExpr) order.SortKeys {
+	var sortKeys order.SortKeys
+	for _, f := range sortExprs {
+		s := a.semSortExpr(f)
+		if this, ok := s.Key.(*dag.This); !ok {
+			// Check not BadExpr so we do not double report errors.
+			if _, ok := s.Key.(*dag.BadExpr); !ok {
+				a.error(f.Expr, fmt.Errorf("field required in sort expression"))
+			}
+		} else {
+			sortKeys = append(sortKeys, order.NewSortKey(s.Order, this.Path))
 		}
```

Comment:
Nit: This is confusing.
```suggestion
		switch key := s.Key.(type) {
		case *dag.This:
			sortKeys = append(sortKeys, order.NewSortKey(s.Order, key.Path))
		case *dag.BadExpr:
			a.error(f.Expr, fmt.Errorf("field required in sort expression"))
		}
```

Reply:
This doesn't work since we're checking to see if key is not but I like the idea of using a switch to approach this.